daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -303,7 +311,8 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
       contractKeyUniqueness = config.contractKeyUniqueness,
       limits = config.limits,
     )
-    interpretLoop(machine, ledgerTime)
+    val discTable = Engine.buildDiscTable(disclosures)
```

Comment:
maybe worth adding a TODO pointing to https://github.com/digital-asset/daml/issues/13864?

Reply:
@rautenrieth-da do you detect duplicate contract ids or key hashes in the participant or should those also be caught here?

Reply:
added.